cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -284,11 +284,11 @@ protected static boolean validFieldName(String fieldName) {
      */
     @SuppressWarnings("unchecked")
     protected static boolean indexLimitReached(Index index, Map<String, Object> existingIndexes) {
-        if (index.indexType.equalsIgnoreCase(Index.TEXT_TYPE)) {
+        if (index.indexType == IndexType.TEXT) {
             for (String name : existingIndexes.keySet()) {
                 Map<String, Object> existingIndex = (Map<String, Object>) existingIndexes.get(name);
                 String type = (String) existingIndex.get("type");
-                if (type.equalsIgnoreCase(Index.TEXT_TYPE) &&
+                if (IndexType.enumValue(type) == IndexType.TEXT &&
```

Comment:
I was confused because the `existingIndexes` structure looks like some sort of deserialised JSON! Well I think that `listIndexesInDatabase` and similar methods should return more self-describing data types, and this should include the use of the Enum.

This way you can limit the conversion between enums and strings to the database specific code in `IndexManager`.